src/solc_0.7/catalyst/AssetAttributesRegistry.sol

```diff
@@ -171,10 +172,12 @@ contract AssetAttributesRegistry is WithMinter, WithUpgrader {
         uint64 blockNumber
     ) internal {
         require(msg.sender == _minter || msg.sender == _upgrader, "NOT_AUTHORIZED_MINTER");
+        // TODO: verify audit comment
         require(gemIds.length <= MAX_NUM_GEMS, "GEMS_MAX_REACHED");
         uint8 maxGems = _gemsCatalystsRegistry.getMaxGems(catalystId);
         require(gemIds.length <= maxGems, "GEMS_TOO_MANY");
         uint16[MAX_NUM_GEMS] memory gemIdsToStore;
+        // TODO: verify audit comment
         for (uint8 i = 0; i < gemIds.length; i++) {
```

Comment:
generally, the length attribute for dynamic calldata arrays should not be trusted, as it is being encoded in the calldata.
More specifically, the user might submit an array.length that is less than the true value. In this case, it does not seem like an issue, but it would be great to have more people looking on it.